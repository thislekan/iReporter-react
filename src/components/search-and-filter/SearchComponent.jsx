import React from 'react';
import style from '../../styles/SearchReports.css';

const SearchFeature = () => (
  <div className={style['filter-div']}>
    <label htmlFor="filter list">Search Incident</label>
    <select name="filter list" id="filter-options">
      <option value="">Select option</option>
      <option value="id">Incident ID</option>
      <option value="location">Location</option>
    </select>
    <input
      type="text"
      name=""
      id="filter-value"
      placeholder="Please enter your search term here"
    />
    <div className={style['search-btns']}>
      <button className={style['filter-list-btn']}>Search</button>
      <button className={style['cancel-search-btn']}>Clear Search</button>
    </div>
  </div>
);

export default SearchFeature;
