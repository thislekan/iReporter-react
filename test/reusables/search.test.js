import React from 'react';
import { shallow } from 'enzyme';
import SearchComponent from '../../src/components/search-and-filter/SearchComponent.jsx';

describe('<SearchComponent />', () => {
  it('Should render the component', () => {
    const wrapper = shallow(
      <SearchComponent />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
