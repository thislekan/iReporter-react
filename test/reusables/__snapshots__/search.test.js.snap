// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchComponent /> Should render the component 1`] = `
<div>
  <label
    htmlFor="filter list"
  >
    Search Incident
  </label>
  <select
    id="filter-options"
    name="filter list"
  >
    <option
      value=""
    >
      Select option
    </option>
    <option
      value="id"
    >
      Incident ID
    </option>
    <option
      value="location"
    >
      Location
    </option>
  </select>
  <input
    id="filter-value"
    name=""
    placeholder="Please enter your search term here"
    type="text"
  />
  <div>
    <button>
      Search
    </button>
    <button>
      Clear Search
    </button>
  </div>
</div>
`;
